### PR TITLE
Upgrade to CVXPY >= 1.1.13.

### PR DESCRIPTION
**Pull request by bettbra**
A collection of mods while upgrading to CVXPY >= 1.1.13:
- Require python >= 3.6 (no harm since CVXPY 1.1 has that same requirement).
- Modify the nonconvex variables to take shape tuples, instead of the previous row, col ints. The new behavior is more consistent with CVXPY 1.
- Some minor PEP 8 formatting mods.
- Ensure all the tests and examples are working.
- Point to a new [scsprox fork.](https://github.com/bettbra/scsprox)

There's more to do to be sure, including sprucing up various docstrings. But that can wait for a separate pull request; seems more useful right now to restore the base to operation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,25 +7,28 @@ NCVX is built on top of [CVXPY](http://www.cvxpy.org/), a domain-specific langua
...
+Then install ``scsprox`` from source [here](https://github.com/bettbra/scsprox).
 
-The easiest way to install the package is to run ``pip install ncvx``. To install the package from source, run ``python setup.py install`` in the main folder. The package has CVXPY and lap as dependencies.
+The easiest way to install the package is to run ``pip install ncvx``. To install the package from source, run ``python setup.py install`` in the main folder.
 
 Example
 -------
 The following code uses NC-ADMM heuristic to approximately solve a least-squares problem where the variable has only ``k`` nonzero components, and all components are between -1 and 1.
 ```
+from cvxpy import sum_squares, sum, Problem, Minimize
```
Comment (SteveDiamond) [inline README.md]: We tend to do ``import cvxpy as cp`` these days. Then you would have ``cp.sum_squares``, etc.

File: examples/circle_packing.py [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
 x_vals = [cp.Variable() for i in range(N)]
 y_vals = [cp.Variable() for i in range(N)]
 
-objective = cp.Minimize(cp.maximum( *(x_vals + y_vals) ) + 0.5)
+objective = cp.Minimize(cp.maximum(*(x_vals + y_vals)) + 0.5)
 constraints = []
 for i in range(N):
-    constraints  += [0.5*r[i] <= x_vals[i],
-                     0.5*r[i] <= y_vals[i]]
+    constraints += [r[i] <= x_vals[i],
```
Comment (SteveDiamond) [inline examples/circle_packing.py:20]: What happened to the 0.5*?

File: examples/circle_packing.py [hunk 3]
```diff
@@ -1,36 +1,40 @@
...
+                    r[i] <= y_vals[i]]
 diff_vars = []
-for i in range(N-1):
-    for j in range(i+1,N):
+for i in range(N - 1):
+    for j in range(i + 1, N):
         t = cp.Variable()
-        diff_vars.append(ncvx.Annulus(2, 0.5 * (r[i]+r[j]), N))
+        diff_vars.append(nc.Annulus(2, r[i] + r[j], N))
```
Comment (SteveDiamond) [inline examples/circle_packing.py:26]: What happened to the 0.5*?

File: examples/factor_model.py [hunk 4]
```diff
@@ -1,48 +1,51 @@
-from cvxpy import *
-from ncvx import *
+import cvxpy as cp
+from cvxpy import norm, Variable, diag, sum_squares, Minimize, trace, Problem, Parameter, pos
```
Comment (SteveDiamond) [inline examples/factor_model.py]: Use ``cp.Variable`` etc instead of separately importing.

File: examples/factor_model.py [hunk 5]
```diff
@@ -1,48 +1,51 @@
...
 def polish_func(sltn):
     matrix = sltn[Sigma_lr.id]
     w, V = np.linalg.eigh(matrix)
     w_sorted_idxs = np.argsort(-w)
     pos_w = w[w_sorted_idxs[:k]]
-    pos_V = V[:,w_sorted_idxs[:k]]
+    pos_V = V[:, w_sorted_idxs[:k]]
     Sigma_tmp = Variable((k, k), symmetric=True)
-    Sigma_small = pos_V*Sigma_tmp*pos_V.T
+    Sigma_small = pos_V * Sigma_tmp * pos_V.T
```
Comment (SteveDiamond) [inline examples/factor_model.py]: Shouldn't this be @ multiplication?

**Comment by SteveDiamond** [comment]
Thanks @bettbra this is a really great PR! I made some minor notes in my review. Generally looks good to go.

**Comment by SteveDiamond** [comment]
Did you run ``flake8`` on it? That's what we use in cvxpy for linting. I would like to switch to ``black`` autoformatting, but that should be a separate PR.

File: README.md (hunk 1 above)
Comment (bettbra) [reply README.md]: I've fixed the README example to have the more standard cvxpy import.

File: examples/circle_packing.py (hunk 2 above)
Comment (bettbra) [reply examples/circle_packing.py:20]: Seems like the original code was setting a circle diameter in the variable ``r``, then multiplying by 0.5 where a radius was called for.  Between renaming the variable to ``d`` or just removing the various 0.5 multiplies, the latter seemed reasonable.

File: examples/circle_packing.py (hunk 3 above)
Comment (bettbra) [reply examples/circle_packing.py:26]: Same as one above; I think the original code was using a diameter in the ``r`` variable.

File: examples/factor_model.py (hunk 4 above)
Comment (bettbra) [reply examples/factor_model.py]: I was shortening a few examples down with the separate ``from``, but I'll remove those in any examples where I used it.

File: examples/factor_model.py (hunk 5 above)
Comment (bettbra) [reply examples/factor_model.py]: Ah, nice, yes it should be. Let me change that.

**Comment by bettbra** [comment]
I haven't fixed all the flake8 issues yet but will later this evening. Gotta take the kids to football right now, ha!

**Comment by SteveDiamond** [comment]
Sounds good! Actually if it's easier you're welcome to just run "black *" to get all the formatting right, since I've reviewed the content. No need to fiddle with flake8 (unless you prefer it).

**Comment by SteveDiamond** [comment]
I decided to merge and apply black formatting myself, just for simplicity.

**Pull request merged**